exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/DatabaseDialect.kt

```diff
@@ -202,6 +209,26 @@ interface DatabaseDialect {
         return TransactionManager.current().db.metadata { resolveReferenceOption(refOption.toString())!! }
     }
 
+    /** Returns a map of all the columns' names mapped to their type. */
+    fun fetchAllColumnTypes(tableName: String): ConcurrentHashMap<String, String> = ConcurrentHashMap()
```

Comment:
Some databases, like H2, MySQL, and MariaDB, have support for the query `SHOW COLUMNS FROM table_name` which provides information about columns and their types. For those databases, it's better to use this query and get all the information in one go, rather than querying the information column by column.

Reply:
I'm not sure if we should return `ConcurrentHashMap<String, String>` here. Could you point me where is? the modification of the result is required, especially the concurrent one?

Reply:
You're right. I thought this would be necessary at the beginning but it later turned out not to be, and I forgot to change it to simply `Map`. Thank you for the comment.